### DISCUSSION TITLE
Deleting old backups files

### DISCUSSION THREAD

**Discussion by silverbullet07** [olahallengren/sql-server-maintenance-solution#596] [General]
What would make the deletion of old backup files just stop?  I have been doing some testing with encrypted backups and when I changed the certificated it seemed to have stopped deleting the old backups. I deleted the whole directory and it started deleting them again. 

I would not think that would have caused them to stop but they just stopped out of the blue. I will be doing some more next week to see if that was the issue or not but was wondering what could have made them stop deleting.

**Comment by olahallengren** [comment]
DatabaseBackup is using xp_delete_file to delete old backup files. I believe that is doing some check to verify that it actually is a SQL Server backup file. I am not sure how that it works with encrypted backups (and specifically when the certificate changes).

Have you been able to reproduce the issue?

**Reply by mlpace54** [reply]
I just installed MaintenanceSolution.sql on SQLSERVER EXPRESS 2017 on my local machine and cannot get dbo.xp_delete_file to work at all in SSMS using the command in the database backup log file. I granted full control to the NT Service\MSSQL$SQLEXPRESS and tried every suggestion I could find in Google posts. I admit that the problem is not with Ola's script but with the dbo.xp_delete_file command not working. Any troubleshooting tips would be appreciated. If I cannot resolve this, I will have to comment out the sections of dbo.DatabaseBackup SP that delete old backups and use a PowerShell script instead. Ola, I sent a more detailed explanation of what I have tried to your direct email address. I realize that dbo.xp_delete_file is not your problem, but perhaps you could recommend troubleshooting tips or the best place to post my issue. Thanks in advance.
